errors.go

```diff
@@ -317,19 +317,22 @@ type ConnectionProblemError struct {
 
 // Error is debug - friendly error message
 func (c *ConnectionProblemError) Error() string {
-	if c.Err == nil {
+	if c.Message != "" {
 		return c.Message
 	}
-	return c.Err.Error()
+	return UserMessage(c.Err)
 }
 
 // IsConnectionProblemError indicates that this error is of ConnectionProblemError type
 func (c *ConnectionProblemError) IsConnectionProblemError() bool {
 	return true
 }
 
-// OrigError returns original error (in this case this is the error itself)
+// OrigError returns original error
 func (c *ConnectionProblemError) OrigError() error {
```

Comment:
I missed some of these errors in my previous PR.
Can you also add `Unwrap() error` methods on the 3 error types here that wrap other errors?